### PR TITLE
Issue 12: Sidekick blocks library config

### PR DESCRIPTION
**Pull request by tmathern**
## Issue

Fix https://github.com/hlxsites/sgws-creative/issues/12

## 🔗 Test URLs:

- Before: https://main--sgws-creative--hlxsites.hlx.page/block-library/library
- After: https://issue-12-library-for-sidekick--sgws-creative--hlxsites.hlx.page/block-library/library

## 📝 Description:

Add blocks library.

### CODE REVIEW

File: tools/sidekick/config.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "project": "Souther Glazers Bloomin' Brands",
```
Comment (arumsey) [inline tools/sidekick/config.json:2]: The project is for `Southern Glazers Creative`. Not just `Bloomin' Brands`.

File: blocks/library-config/library-config.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import { createTag } from '../../scripts/scripts.js';
+
+const validListTypes = ['blocks', 'sections', 'buttons', 'placeholders', 'assets', 'templates'];
+
+const LIBRARY_PATH = '/careers/block-library/library.json';
```
Comment (arumsey) [inline blocks/library-config/library-config.js:5]: Path needs updating

File: blocks/library-config/library-config.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  const list = createLibraryList(libraries);
+  skLibrary.append(list);
+  el.attachShadow({ mode: 'open' });
+
+  el.shadowRoot.append(skLibrary);
+
+  // add styles
+  const link = document.createElement('link');
+  link.setAttribute('rel', 'stylesheet');
+  link.setAttribute('href', '/careers/blocks/library-config/library-config.css');
```
Comment (arumsey) [inline blocks/library-config/library-config.js:152]: Update path

File: tools/sidekick/config.json (hunk 1 above)
Comment (tmathern) [reply tools/sidekick/config.json]: Committed your suggested change, @arumsey.

File: tools/sidekick/config.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+            "title": "Library",
+            "environments": [ "edit" ],
+            "isPalette": true,
+            "paletteRect": "top: auto; bottom: 20px; left: 20px; height: 398px; width: 360px;",
+            "url": "/block-library/library",
+            "includePaths": [ "**.docx**" ]
+        },
+        {
+            "id": "publish",
+            "excludePaths": [ "**/drafts/**","**%2Fdrafts**" ]
```
[the remark this answers is not in the document]
Comment (tmathern) [reply tools/sidekick/config.json:16]: Probably to fix an encoding issue from the latest iterations of that block config, as `"%2f" becomes an ordinary character that simply represents "/" character in element of your url`.

**Review by arumsey — APPROVED** [review]

**Comment by arumsey** [comment]
👍

**Pull request merged**